### DISCUSSION TITLE
Increase performance with Offscreen canvas

### DISCUSSION THREAD

**Discussion by eltaurito** [fabricjs/fabric.js#8800] [Q&A] [answered]
Hi all guys.

I open this discussion because I would like to find a way to increase the performances of my application by using the offscreen canvas feature.

https://developer.chrome.com/blog/offscreen-canvas/

I have built my app with [Electron Rect Boilerplate](https://electron-react-boilerplate.js.org/), and basically what it does is drawing a big number (around 15000) of simple [Rects ](http://fabricjs.com/docs/fabric.Rect.html). 

The app does its job, but the problem is that it's taking too many seconds to render. 
While it's rendering, the main thread is stuck, so this means that the UI is completely freezed until the rendering is completed.

I saw some examples of the offscreen-canvas usage, and they were pretty impressive.

https://devnook.github.io/OffscreenCanvasDemo/index.html

https://threejs.org/examples/webgl_worker_offscreencanvas.html

So I would like to know if someone of you knows how to integrate this feature in fabricjs.

Thanks a lot.

Davide

**Comment by ShaMan123** [comment, marked as answer]
Always a good idea to check PRs
Offscreen canvas: #8774 
Caching strategies: https://github.com/fabricjs/fabric.js/discussions/7874#discussioncomment-2541974 #8581

**Reply by eltaurito** [reply]
Thank you @ShaMan123 for your answer.

Sorry if i didn't check PRs. 

However, in order to interact as best as possibile with this community, I've read the [CONTRIBUTING GUIDE](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md).

**Reply by ShaMan123** [reply]
All is well
I think we should make it possible to use offscreen w/o workarounds.
It is a matter of moving browser specific code to the right place. If you are up for it I can back you and point out where to start.
